playground/src/editor/editor.js

```diff
@@ -184,4 +188,33 @@ class Editor {
     return this.codeMirror.getTokenAt(pos, precise);
   }
 
+  loadHints(validator) {
+    this.amphtmlHints.then((hints) => {
+      for (let key of Object.keys(CodeMirror.htmlSchema)) {
+        delete CodeMirror.htmlSchema[key];
+      }
+      Object.assign(CodeMirror.htmlSchema, hints[validator]);
+    });
+  }
+
+  setRuntime(runtime) {
+    const runtimeData = runtimes.get(runtime.id);
+    this.loadHints(runtimeData.validator);
+  }
+
+  fetchHintsData() {
+    return new Promise((resolve, reject) => {
+      window.requestIdleCallback(() => {
```

Comment:
note to myself: I should create a promised version for requestIdleCallback so that we can rewrite this as 

```
runWhenIdle()
     .then( () => fetch(HINTS_URL))
     .then( (response) => response.json() )
```

Reply:
Good idea, can be a one-liner: `runWhenIdle = (options) => new Promise((resolve) => window.requestIdleCallback(resolve, options))`

Would've actually simplified this PR!